packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -87,26 +87,23 @@ enum _InitialResult { crashed, timedOut, connected }
 enum _TestResult { crashed, harnessBailed, testBailed }
 typedef Future<Null> _Finalizer();
 
-class CompilationRequest {
+class _CompilationRequest {
   String path;
   Completer<String> result;
 
-  CompilationRequest(this.path, this.result);
+  _CompilationRequest(this.path, this.result);
 }
 
-class _FlutterPlatform extends PlatformPlugin {
-  _FlutterPlatform({
-    @required this.shellPath,
-    this.watcher,
-    this.enableObservatory,
-    this.machine,
-    this.startPaused,
-    this.explicitObservatoryPort,
-    this.host,
-    this.previewDart2,
-    this.port,
-    this.dillFilePath,
-  }) : assert(shellPath != null) {
+// This class is a wrapper around compiler that allows multiple isolates enqueue
+// compilation requests, but ensures only one compilation at a time.
+class _Compiler {
+  final StreamController<_CompilationRequest> compilerController;
+  ResidentCompiler compiler;
+  final List<_CompilationRequest> compilationQueue;
+
+  _Compiler() :
```

Comment:
Put the constructor first

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#constructors-come-first-in-a-class